### DISCUSSION TITLE
Can a Device inherit tags set on the Device_Type

### DISCUSSION THREAD

**Discussion by kevintedder** [netbox-community/netbox#7618] [Q&A]
As a NOOB to Netbox and as the title suggests, I've tried setting Tags on the Device_Type, but when I create a Device using it, the Tags do not get inherited.   Is this by design or have it done something wrong?

One idea is to create a set of Tags that describe additional functionality/capability over and above the Device_Role.  Once the Tags are set on the Device_Type, all Devices created will get the same Tags. We can then define a greater level of granularity to  search on other than just the Device_Role.

Eg. A Cisco Cat6500 can do both L2_switching and L3_routing (Device_Type Tags) if so configured. But the Device_Role defines where in the datacentre it is installed - i.e. Front End, Back End or DMZ, etc...  Whilst some Device_Types may only have L2-switching in the same Device_Role (DC area) it would allow us to search for all devices in the DMZ that perform L3_routing.

We are also looking at other, similar, search mechanisms using Tags. So this would be a powerful feature for us.

**Comment by jeremystretch** [comment]
> when I create a Device using it, the Tags do not get inherited

This is by design. Tags may be used for any purpose, and their application to a device type may be entirely unrelated to device instances, so NetBox can't assume that you want to replicate the assigned tags.

**Reply by kevintedder** [reply]
@jeremystretch - Thx for the quick response. 
Ok I understand.     Can I then ask why the Device_Type allows us to input Tags if they are not going to be used.    Would a better option be to inherit the set Tags set by the administrator, then allow the user to remove those that are not required for each individual device.
I assume that the Device_Type is a template to enforce the device to be of a specific setup. Then be tailored/modified afterwards.

**Reply by jeremystretch** [reply]
> Can I then ask why the Device_Type allows us to input Tags if they are not going to be used.

I think you missed my point. There are plenty of likely use cases for using tags to manage device types themselves, just as with all the other models in NetBox that support the assignment of tags.

**Reply by kevintedder** [reply]
@jeremystretch -  Ah. I think the penny has dropped.   In the DB the device_type object Tags are available as they are linked through to the Device object.  Thanks for your response.
